fn int foo()
{
    return 1;
}

fn void test()
{
    int x = foo();
    if (x > 0) return;
    unreachable();
    x++;
}


/* #expect: unreachable.ll


define void @unreachable.test() #0 {
entry:
  %x = alloca i32, align 4
  %0 = call i32 @unreachable.foo()
  store i32 %0, ptr %x, align 4
  %1 = load i32, ptr %x, align 4
  %gt = icmp sgt i32 %1, 0
  br i1 %gt, label %if.then, label %if.exit

if.then:                                          ; preds = %entry
  ret void

if.exit:                                          ; preds = %entry
  call void @"std::core::builtin.panic"(ptr @.str, ptr @.str.1, ptr @.str.2, i32 10)
  unreachable

after.unreachable:                                ; No predecessors!
  %2 = load i32, ptr %x, align 4
  %add = add i32 %2, 1
  store i32 %add, ptr %x, align 4
  ret void
}

